src/routes/_surfaces/inspector/SourceInspector.svelte

```diff
@@ -0,0 +1,229 @@
+  import ColumnProfileNavEntry from "$lib/components/column-profile/ColumnProfileNavEntry.svelte";
+  import Explore from "$lib/components/icons/Explore.svelte";
+  import Model from "$lib/components/icons/Model.svelte";
+  import { GridCell, LeftRightGrid } from "$lib/components/left-right-grid";
+  import Tooltip from "$lib/components/tooltip/Tooltip.svelte";
+  import TooltipContent from "$lib/components/tooltip/TooltipContent.svelte";
+  import {
+    autoCreateMetricsDefinitionForSource,
+    createModelForSource,
+  } from "$lib/redux-store/source/source-apis";
+  import { selectTimestampColumnFromProfileEntity } from "$lib/redux-store/source/source-selectors";
+  import {
+    formatBigNumberPercentage,
+    formatInteger,
+  } from "$lib/util/formatters";
+  import { getContext } from "svelte";
+  import { slide } from "svelte/transition";
+
+  const persistentModelStore = getContext(
+    "rill:app:persistent-model-store"
+  ) as PersistentModelStore;
+
+  const persistentTableStore = getContext(
+    "rill:app:persistent-table-store"
+  ) as PersistentTableStore;
+  const derivedTableStore = getContext(
+    "rill:app:derived-table-store"
+  ) as DerivedTableStore;
+
+  const store = getContext("rill:app:store") as ApplicationStore;
+
+  let showColumns = true;
+
+  /** Select the explicit ID to prevent unneeded reactive updates in currentTable */
+  $: activeEntityID = $store?.activeEntity?.id;
+
+  let currentTable: PersistentTableEntity;
+  $: currentTable =
+    activeEntityID && $persistentTableStore?.entities
+      ? $persistentTableStore.entities.find((q) => q.id === activeEntityID)
+      : undefined;
+  let currentDerivedTable: DerivedTableEntity;
+  $: currentDerivedTable =
+    activeEntityID && $derivedTableStore?.entities
+      ? $derivedTableStore.entities.find((q) => q.id === activeEntityID)
+      : undefined;
+  // get source table references.
+
+  // toggle state for inspector sections
+
+  $: timestampColumns =
+    selectTimestampColumnFromProfileEntity(currentDerivedTable);
+
+  const handleCreateModelFromSource = async () => {
+    await createModelForSource(
+      $persistentModelStore.entities,
+      currentTable.tableName
+    );
+  };
+
+  const handleCreateMetric = () => {
+    // A side effect of the createMetricsDefsApi is we switch active assets to
+    // the newly created metrics definition. So, this'll bring us to the
+    // MetricsDefinition page. (The logic for this is contained in the
+    // not-pictured async thunk.)
+    autoCreateMetricsDefinitionForSource(
+      $persistentModelStore.entities,
+      $derivedTableStore.entities,
+      currentTable.id,
+      $persistentTableStore.entities.find(
+        (table) => table.id === activeEntityID
+      ).tableName
+    );
+  };
+
+  /** source summary information */
+  let sourceType;
+  let rowCount;
+  let columnCount;
+  let nullPercentage;
+  $: {
+    switch (currentTable?.sourceType) {
+      case 0: {
```

Comment:
Just in case the Enum ordering changes, what about this format: `case TableSourceType.ParquetFile:`